dbms/src/Storages/LiveView/StorageLiveView.cpp

```diff
@@ -177,11 +229,25 @@ void StorageLiveView::writeIntoLiveView(
         }
     }
 
-    auto parent_storage = context.getTable(live_view.getSelectDatabaseName(), live_view.getSelectTableName());
-    auto proxy_storage = std::make_shared<ProxyStorage>(parent_storage, std::move(from), QueryProcessingStage::WithMergeableState);
-    InterpreterSelectQuery select(live_view.getInnerQuery(), context, proxy_storage, QueryProcessingStage::Complete);
+    auto blocks_storage = StorageBlocks::createStorage(live_view.database_name, live_view.table_name, parent_storage->getColumns(), std::move(from), QueryProcessingStage::WithMergeableState);
+    InterpreterSelectQuery select(mergeable_query->clone(), context, blocks_storage, SelectQueryOptions(QueryProcessingStage::Complete));
     BlockInputStreamPtr data = std::make_shared<MaterializingBlockInputStream>(select.execute().in);
 
+    if (live_view.getInnerSubQuery())
+    {
+        auto outer_blocks_storage = StorageBlocks::createStorage("_liveview", live_view.table_name + "_blocks", ColumnsDescription(data->getHeader().getNamesAndTypesList()), {data}, QueryProcessingStage::FetchColumns);
+        block_context->addExternalTable(live_view.table_name + "_blocks", outer_blocks_storage);
+        try
+        {
+            InterpreterSelectQuery outer_select(live_view.getInnerOuterQuery(), *block_context, SelectQueryOptions(QueryProcessingStage::Complete));
+            data = std::make_shared<MaterializingBlockInputStream>(outer_select.execute().in);
+        }
+        catch (...)
+        {
+            block_context->tryRemoveExternalTable(live_view.table_name + "_blocks");
```

Comment:
But we don't rethrow exception here.
And also I don't understand why we need to remove this table. `block_context` is a local for function, and will be destroyed anyway.

Reply:
Ok, I will remove this. Like you said, block_context will be destroyed anyway.